Plan/react/dashboard/src/hooks/context/alertPopupContext.js

```diff
@@ -0,0 +1,32 @@
+import {createContext, useCallback, useContext, useMemo, useState} from "react";
+
+const AlertPopupContext = createContext({});
+
+export const AlertPopupContextProvider = ({children}) => {
+    const [alerts, setAlerts] = useState([]);
+
+    const dismissAlert = useCallback(alert => {
+        setAlerts(alerts.filter(a => a.time - alert.time < 1));
+    }, [alerts, setAlerts]);
+
+    const addAlert = useCallback(alert => {
+        const time = Date.now() + Math.random();
+        const addedAlert = {time, ...alert};
+        setAlerts([...alerts, addedAlert]);
+        setTimeout(() => {
+            dismissAlert(addedAlert);
```

Comment:
Dismissing multiple alerts is buggy and the timer animation doesn't work properly when one alert is dismissed